pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_runtime_reconcile.go

```diff
@@ -165,6 +174,18 @@ func (r *runtimeReconcile) installSingletonAgent(ctx context.Context, klusterlet
 		return klusterlet, reconcileStop, err
 	}
 
+	// clean registration/work if there is any
```

Comment:
do we need to clean the other resources like role/rolebindings?

Reply:
they are shared and will be updated accordingly when mode is changed, so not need to remove.